### ISSUE TITLE
Failed to create CUDAExecutionProvider.

### ISSUE THREAD

**Issue by jFkd1** [microsoft/onnxruntime#17537]
### Describe the issue

For some reason, onnxruntime-gpu is having a hard time using CUDAExecutionProvider. 

Using Cuda 11.7 with onnxruntime-gpu=1.15.1, I tried running the following in python 3.10:

```
import torch
import onnxruntime
save_path = 'model.onnx'
ort_session = onnxruntime.InferenceSession(save_path, providers=['CUDAExecutionProvider'])
```

And I will always get the error message: 
> [W:onnxruntime:Default, onnxruntime_pybind_state.cc:640 CreateExecutionProviderInstance] Failed to create CUDAExecutionProvider. Please reference https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements to ensure all dependencies are met.

It looks like I am using the right system and versions. I will upload the model onnx if necessary. 

### To reproduce

pip install onnxruntime-gpu==1.15.1

Using cuda 11.7

Run this code: 
```
import torch
import onnxruntime
save_path = 'model.onnx'
ort_session = onnxruntime.InferenceSession(save_path, providers=['CUDAExecutionProvider'])
```

### Urgency

Very urgent

### Platform

Linux

### OS Version

Ubuntu 18.04.6 LTS

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.15.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

CUDA

### Execution Provider Library Version

CUDA 11.7

**Comment by jFkd1** [comment]
Model download path: https://github.com/facefusion/facefusion-assets/releases/download/models/inswapper_128.onnx

**Comment by tianleiwu** [comment]
It is not related to model. Most likely related to package installation or cuDNN installation. Are you able to reproduce it in a new python environment, or reinstall like the following?

```
pip3 uninstall onnxruntime onnxruntime-gpu ort-nightly ort-nightly-gpu
pip3 install onnxruntime-gpu
pip3 install torch torchvision torchaudio --force-reinstall --index-url https://download.pytorch.org/whl/cu117
```

Then test with your python script
```
import torch
import onnxruntime
save_path = 'model.onnx'
ort_session = onnxruntime.InferenceSession(save_path, providers=['CUDAExecutionProvider'])
```
Since you imported torch before onnxruntime, ORT shall be able to use the cuDNN loaded by torch.

**Comment by snnn** [comment]
You can also try our nightly build.

To get it, first uninstall the one you have installed:
```bash
         python3 -m pip uninstall -y ort-nightly-gpu ort-nightly onnxruntime onnxruntime-gpu -qq
```

Then get a new one from our nightly feed:

```
python3 -m pip install coloredlogs flatbuffers numpy packaging protobuf sympy
python3 -m pip install -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ ort_nightly
```

Then you can use the `ldd` tool to figure out which library was missing.

**Comment by jFkd1** [comment]
hi @tianleiwu , thanks for commenting. I tried reinstall it but it does not seem to help. I have also tried this on a different server (with the exact same cuda/python environment) but am still getting the same error. 

@snnn thanks for the suggestion. Looks like the nightly build also is not working properly... The only difference I see is just that the error output is now colored yellow. 

Can you specify on what you mean by using `ldd`?

**Comment by snnn** [comment]
To where the package is installed, you may find some *.so files. For example, if you run 
```
find  . -name \*.so -exec ldd {} \;
```
in the directory where ort-nightly-gpu python package is installed, the ldd tool should show some errors.

**Comment by snnn** [comment]
It seems totally fine. As [tianleiwu](https://github.com/tianleiwu) said, the "import torch" statement could be the reason. If you load pytorch before loading onnxruntime, onnxruntime probably will get cuda and cudnn libs from pytorch instead and there might be version mismatch happened.  Can you try to create a simpler script that only has onnxruntime?

**Comment by snnn** [comment]
The latest nightly package has some problems. But the latest release package is fine. I just tested it under RHEL8 and it worked fine. The running environment has the following packages:
```
cuda-toolkit-11-config-common-11.8.89-1.noarch
cuda-cudart-11-8-11.8.89-1.x86_64
cuda-nvrtc-11-8-11.8.89-1.x86_64
libnccl-2.15.5-1+cuda11.8.x86_64
libcudnn8-8.9.0.131-1.cuda11.8.x86_64
cuda-toolkit-config-common-12.1.105-1.noarch
cuda-toolkit-11-8-config-common-11.8.89-1.noarch
cuda-compat-11-8-520.61.05-1.x86_64
cuda-libraries-11-8-11.8.0-1.x86_64
cuda-nvtx-11-8-11.8.86-1.x86_64
```

**Comment by jFkd1** [comment]
Thanks for the responses. 

I tried not importing torch and that did not seem to mitigate the issue. I was only importing torch because it was believed to solve the CUDA dependency issue with ort for some. Unfortunately it did not work for me. 

Here's what I get from running ` find  . -name \*.so -exec ldd {} \` in my ort-nightly install:

```
/qitian/miniconda3/envs/roop/lib/python3.10/site-packages/onnxruntime/capi$ find  . -name \*.so -exec ldd {} \;
        linux-vdso.so.1 (0x00007ffff7ffb000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007ffff6bf4000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007ffff69ec000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007ffff67cf000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007ffff65b0000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007ffff6227000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007ffff5e89000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007ffff5c71000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ffff5880000)
        /lib64/ld-linux-x86-64.so.2 (0x00007ffff7dd3000)
        linux-vdso.so.1 (0x00007ffff7ffb000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007ffff7848000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007ffff74aa000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007ffff7292000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ffff6ea1000)
        /lib64/ld-linux-x86-64.so.2 (0x00007ffff7dd3000)
Inconsistency detected by ld.so: dl-version.c: 205: _dl_check_map_versions: Assertion `needed != NULL' failed!
Inconsistency detected by ld.so: dl-version.c: 205: _dl_check_map_versions: Assertion `needed != NULL' failed!
```

A quick internet search suggests this may be related to https://github.com/microsoft/onnxruntime/issues/9754. I will try the suggested fix with reinstalling libcudnn8 and see if that would help.

**Comment by snnn** [comment]
I will make a new nightly package for you to test, which will not have the #9754 issue. But I will need a few days.

**Comment by mahesh11T** [comment]
following, 
FAIL : Failed to load library libonnxruntime_providers_cuda.so with error: libcurand.so.10: cannot open shared object file: No such file or directory
#/usr/local/lib/python3.8/site-packages/onnxruntime/capi/libonnxruntime_providers_cuda.so:
issue solved?

**Comment by snnn** [comment]
Right, as it said, your operating system cannot find libcurand.so.10. Did you install CUDA and CUDNN?

**Comment by snnn** [comment]
Feel free to create a new issue if the problem still exists. Please note CUDA and CUDNN are commercial software owned by Nvidia with End User License Agreements. Our team do not redistribute their software due to license restrictions and security concerns. Even if we do, there is no way to prove to you the redistributed files are genuine.  The latest ONNX Runtime release was built with CUDA 11.8 and CUDNN 8.9. Users of ONNX Runtime GPU packages need to get the dependent libraries from Nvidia.  If any dependent library is missing, on Linux ONNX Runtime should be able to print out a detailed error message when loading its CUDA execution provider. The message can tell what was missing. 
A side note: if you need to have multiple CUDA versions in parallel, it's better to not add any of them to ld.so's global database(/etc/ld.so.cache) and you need to add one of them to LD_LIBRARY_PATH environment variable.
